RecoEgamma/PhotonIdentification/src/PhotonMVABasedHaloTagger.cc

```diff
@@ -0,0 +1,424 @@
+
+    double dR2 = reco::deltaR2(phoSCEta, phoSCPhi, rhEta, rhPhi);
+
+    if (dR2 < 0.15 * 0.15 && !isRHBehindECAL) {  ///dont use hits which are just behind the ECAL in the same phi region
+      hcalClusX_samedR_ += rhX * rhE;
+      hcalClusY_samedR_ += rhY * rhE;
+      hcalClusZ_samedR_ += rhZ * rhE;
+      hcalClusE_samedR_ += rhE;
+      hcalClusNhits_samedR_++;
+    }
+
+  }  //for(int ih=0; ih<nMatchedHBHErh[ipho]; ih++)
+
+  if (hcalClusNhits_samedPhi_ > 0) {
+    hcalClusX_samedPhi_ = hcalClusX_samedPhi_ / hcalClusE_samedPhi_;
+    hcalClusY_samedPhi_ = hcalClusY_samedPhi_ / hcalClusE_samedPhi_;
+    hcalClusZ_samedPhi_ = hcalClusZ_samedPhi_ / hcalClusE_samedPhi_;
+  }  //if(hcalClusNhits_samedPhi_>0)
+
+  if (hcalClusNhits_samedR_ > 0) {
+    hcalClusX_samedR_ = hcalClusX_samedR_ / hcalClusE_samedR_;
+    hcalClusY_samedR_ = hcalClusY_samedR_ / hcalClusE_samedR_;
+    hcalClusZ_samedR_ = hcalClusZ_samedR_ / hcalClusE_samedR_;
+  }  //if(hcalClusNhits_samedR_>0)
+}
+
+void PhotonMVABasedHaloTagger::calmatchedESCoordForBothHypothesis(const CaloGeometry* geo,
+                                                                  const reco::Photon* pho,
+                                                                  const EcalRecHitCollection& ESRecHits) {
+  preshowerX_samedPhi_ = 0;
+  preshowerY_samedPhi_ = 0;
+  preshowerZ_samedPhi_ = 0;
+  preshowerNhits_samedPhi_ = 0;
+  preshowerE_samedPhi_ = 0;
+
+  preshowerX_samedR_ = 0;
+  preshowerY_samedR_ = 0;
+  preshowerZ_samedR_ = 0;
+  preshowerNhits_samedR_ = 0;
+  preshowerE_samedR_ = 0;
+
+  double phoSCEta = pho->superCluster()->eta();
+  double phoSCPhi = pho->superCluster()->phi();
+
+  double tmpDiffdRho = 999;
+  double matchX_samephi = -999;
+  double matchY_samephi = -999;
+  bool foundESRH_samephi = false;
+
+  double tmpDiffdRho_samedR = 999;
+  double matchX_samedR = -999;
+  double matchY_samedR = -999;
+  bool foundESRH_samedR = false;
+
+  ///get theta and phi of the coordinates of photon
+  double tan_theta = 1. / sinh(phoSCEta);
+
+  double cos_phi = cos(phoSCPhi);
+  double sin_phi = sin(phoSCPhi);
+
+  for (const auto& esrechit : ESRecHits) {
+    const GlobalPoint& rechitPoint = geo->getPosition(esrechit.id());
+
+    double rhEta = rechitPoint.eta();
+    double rhX = rechitPoint.x();
+    double rhY = rechitPoint.y();
+    double rhZ = rechitPoint.z();
+    double rhE = esrechit.energy();
+
+    if (phoSCEta * rhEta < 0)
+      continue;
+
+    if (rhE < noiseThrES_)
+      continue;
+
+    ////try to include RH according to the strips, 11 in X and 11 in Y
+    /////////First calculate RH nearest in phi and eta to that of the photon SC
+
+    //////same phi ----> the X and Y should be similar
+    double dRho2 = pow(rhX - ecalClusX_, 2) + pow(rhY - ecalClusY_, 2);
+
+    if (dRho2 < tmpDiffdRho &&
+        dRho2 <
+            2.2 *
+                2.2) {  ////i.e. hit is required to be within the ----> seems better match with the data compared to 2.47
```

Comment:
this formatting looks somewhat weird, is this what `code-format` requires?

Reply:
it's a "classic" issue with the code format for comments on the same line with the code: the code-format can not add linebreaks to the comment and does something silly with iterative line breaks for the remaining code.
The solution is to move the comment out of the line of code